x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/ttest/TStatsBuilder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.analytics.ttest;
+
+import org.elasticsearch.common.lease.Releasable;
+import org.elasticsearch.common.lease.Releasables;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.util.DoubleArray;
+import org.elasticsearch.common.util.LongArray;
+import org.elasticsearch.search.aggregations.metrics.CompensatedSum;
+
+public class TStatsBuilder implements Releasable {
+
+    private LongArray counts;
+    private DoubleArray sums;
+    private DoubleArray compensations;
+    private DoubleArray sumOfSqrs;
+    private DoubleArray sumOfSqrCompensations;
+
+    TStatsBuilder(BigArrays bigArrays) {
+        counts = bigArrays.newLongArray(1, true);
+        sums = bigArrays.newDoubleArray(1, true);
+        compensations = bigArrays.newDoubleArray(1, true);
+        sumOfSqrs = bigArrays.newDoubleArray(1, true);
+        sumOfSqrCompensations = bigArrays.newDoubleArray(1, true);
+    }
+
+    public TTestStats get(long bucket) {
+        return new TTestStats(counts.get(bucket), sums.get(bucket), sumOfSqrs.get(bucket));
+    }
+
+    public long build(long bucket) {
+        return counts.get(bucket);
+    }
+
+    public long getSize() {
+        return counts.size();
+    }
+
+    public void grow(BigArrays bigArrays, long buckets) {
+        counts = bigArrays.grow(counts, buckets);
```

Comment:
TBH, I'm not sure how expensive `grow()` is... I don't think it's very expensive so it might not matter.  But some aggs that have a lot of big arrays to manage will call `BigArrays#overSize()` method directly and then resize each of their arrays, instead of grow'ing each.  

StatsAggregator is a good example: https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/aggregations/metrics/StatsAggregator.java#L90-L95